drivers/gnss/gnss_quectel_lx6.c

```diff
@@ -0,0 +1,758 @@
+		return ret;
+	}
+
+	ret = quectel_lx6_configure_pps(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to configure PPS: %d", ret);
+		modem_pipe_close(data->pipe, K_SECONDS(10));
+		return ret;
+	}
+
+	return ret;
+}
+
+static int quectel_lx6_suspend(const struct device *dev)
+{
+	const struct quectel_lx6_config *config = dev->config;
+	struct quectel_lx6_data *data = dev->data;
+	int ret;
+
+	LOG_DBG("Suspend");
+
+	if (config->vcc.port != NULL) {
+		ret = gpio_pin_configure_dt(&config->vcc, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("[%s] VCC pin active", dev->name);
+		}
+	} else {
+		/* TODO add check that either VCC or FORCE_ON are defined */
+		return -ENOTSUP;
+	}
+
+	modem_pipe_close(data->pipe, K_SECONDS(10));
+
+	return ret;
+}
+
+#if CONFIG_GNSS_QUECTEL_LX6_RESET_ON_INIT
+static int quectel_lx6_gpio_reset(const struct device *dev)
+{
+	const struct quectel_lx6_config *config = dev->config;
+	int ret;
+
+	if ((config->reset.port != NULL) && (config->vcc.port != NULL)) {
+		ret = gpio_pin_set_dt(&config->vcc, 1);
+		if (ret < 0) {
+			LOG_ERR("[%s] couldn't config VCC", dev->name);
+			return ret;
+		}
+		ret = gpio_pin_set_dt(&config->reset, 0); /* Inactive is high */
+		if (ret < 0) {
+			LOG_ERR("Failed to inactivate reset pin: %d", ret);
+		}
+		k_msleep(4);                              /* > 2ms */
+		ret = gpio_pin_set_dt(&config->reset, 1); /* Active is low */
+		if (ret < 0) {
+			LOG_ERR("Failed to activate reset pin: %d", ret);
+		}
+		k_msleep(12);                             /* Pulldown > 10ms */
+		ret = gpio_pin_set_dt(&config->reset, 0); /* Inactive is high */
+		if (ret < 0) {
+			LOG_ERR("Failed to inactivate reset pin: %d", ret);
+		}
+	} else {
+		LOG_WRN("[%s] couldn't reset", dev->name);
+	}
+
+	return ret;
+}
+#endif
+
+static int quectel_lx6_turn_on(const struct device *dev)
+{
+	const struct quectel_lx6_config *config = dev->config;
+	struct quectel_lx6_data *data = dev->data;
+	int ret;
+
+	LOG_INF("Turn on");
+
+	if (config->vcc.port != NULL) {
+		ret = gpio_pin_configure_dt(&config->vcc, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("[%s] couldn't activate VCC", dev->name);
+			return ret;
+		}
+	}
+
+#if CONFIG_GNSS_QUECTEL_LX6_RESET_ON_INIT
+	if (!data->oneshot_reset) {
+		data->oneshot_reset = true;
+		quectel_lx6_gpio_reset(dev);
+	}
+#endif
+
+	k_msleep(250);
+
+	ret = modem_pipe_open(data->pipe, K_SECONDS(10));
+	if (ret < 0) {
+		LOG_ERR("Failed to open modem pipe: %d", ret);
+	}
+
+	return ret;
+}
+
+static int quectel_lx6_turn_off(const struct device *dev)
+{
+	const struct quectel_lx6_config *config = dev->config;
+	struct quectel_lx6_data *data = dev->data;
+	int ret;
+
+	LOG_INF("Turn off");
+
+	if (config->vcc.port != NULL) {
+		ret = gpio_pin_configure_dt(&config->vcc, GPIO_INPUT);
+		if (ret < 0) {
+			LOG_ERR("[%s] couldn't avoid back powering VCC", dev->name);
+			return ret;
+		}
+	}
+
+	return modem_pipe_close(data->pipe, K_SECONDS(10));
```

Comment:
You may want to close the pipe before disconnecting vcc, if that actually powers down the modem you could get garbage data on the UART or frame errors as the UART bus may be "floating"